shm/src/Ship.cpp

```diff
@@ -0,0 +1,174 @@
+#include "shm/inc/Ship.hpp"
+
+#include <algorithm>
+#include <cstddef>
+#include <iomanip>
+#include <iostream>
+#include <stdexcept>
+
+#include "shm/inc/Cargo.hpp"
+#include "shm/inc/Delegate.hpp"
+#include "shm/inc/Player.hpp"
+#include "shm/inc/Store.hpp"
+
+Ship::Ship(int id, const std::string& name, size_t speed, size_t maxCrew, size_t capacity, Time* time, Delegate* delegate)
+    : id_(id)
+    , name_(name)
+    , speed_(speed)
+    , maxCrew_(maxCrew)
+    , capacity_(capacity)
+    , time_(time)
+    , delegate_(delegate)
+{
+    time_->addSubscriber(this);
```

Comment:
Should also remove subscriber in D'tor

Reply:
To avoid situations like, your object will be deleted but the subscriber still has a pointer to it.